packages/sit-onyx/src/composables/useCustomValidity.ts

```diff
@@ -61,34 +95,89 @@ export const useCustomValidity = (options: UseCustomValidityOptions) => {
        */
       watchEffect(() => el.setCustomValidity(options.props.customError ?? ""));
 
-      /**
-       * Update validityState ref when the input changes.
-       */
       watch(
-        [() => options.props.customError, () => options.props.modelValue, isTouched],
+        [() => options.props.customError, () => options.props.modelValue],
         () => {
           const newValidityState = transformValidityStateToObject(el.validity);
 
-          // do not emit validityChange event if input is valid and has never been invalid
-          if (!isTouched.value || (!validityState.value && newValidityState.valid)) return;
+          // do not emit update if input is valid and has never been invalid
+          if (!validityState.value && newValidityState.valid) return;
 
           // ignore if actual validity state value is unchanged
           if (validityState.value && areObjectsFlatEqual(newValidityState, validityState.value)) {
             return;
           }
 
           validityState.value = newValidityState;
+        },
+        { immediate: true },
+      );
+
+      /**
+       * Update validityState ref when the input changes.
+       */
+      watch(
+        [() => options.props.customError, validityState, isDirty],
+        () => {
+          // do not emit validityChange event if the value was never changed
+          if (!isDirty.value || !validityState.value) return;
+
           options.emit("validityChange", validityState.value);
         },
         { immediate: true },
       );
     },
   } satisfies Directive<InputValidationElement, undefined>;
 
+  const errorMessages = computed<FormErrorMessages>(() => {
+    if (!validityState.value || validityState.value.valid) return undefined;
```

Comment:
```suggestion
    if (!validityState.value || validityState.value.valid) return;
```

returning "nothing" is equal to undefined